### PR TITLE
Add more variations of QuickTime file format

### PR DESCRIPTION
**Pull request by karlhiramoto**
https://www.loc.gov/preservation/digital/formats/fdd/fdd000052.shtml
http://www.garykessler.net/library/file_sigs.html

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -206,7 +206,10 @@ module.exports = input => {
 		}
 	}
 
-	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70])) {
+	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70]) ||
+		(buf[4] === 0x66 && buf[5] === 0x72 && buf[6] === 0x65 && buf[7] === 0x65) ||
+		(buf[4] === 0x6D && buf[5] === 0x64 && buf[6] === 0x61 && buf[7] === 0x74) ||
+		(buf[4] === 0x77 && buf[5] === 0x69 && buf[6] === 0x64 && buf[7] === 0x65)) {
```
Comment (sindresorhus) [inline index.js]: Where is this taken from? I couldn't see it in any of your links.

**Comment by sindresorhus** [comment]
You seem to be missing this one:

```
[4 byte offset]
6D 6F 6F 76	 	[4 byte offset]
moov
MOV	 	QuickTime movie file
```

File: index.js [hunk 2]
```diff
@@ -206,7 +206,10 @@ module.exports = input => {
 		}
 	}
 
-	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70])) {
+	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70]) ||
```
Comment (sindresorhus) [inline index.js]: Seems like this one should be stricter:

```
00 00 00 14 66 74 79 70
71 74 20 20	 	....ftyp
qt
MOV	 	QuickTime movie file
```

**Comment by sindresorhus** [comment]
And this:

```
[4 byte offset]
66 74 79 70 71 74 20 20	 	[4 byte offset]
ftypqt
MOV	 	QuickTime movie file
```

**Comment by karlhiramoto** [comment]
@sindresorhus   I made the check more strict in 6ab2699  

I  hesitate to add all the other variations because I don't have these files to test.

File: index.js [hunk 3]
```diff
@@ -206,7 +206,12 @@ module.exports = input => {
 		}
 	}
 
-	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70])) {
+	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70, 0x71, 0x74, 0x20, 0x20]) ||
+		(buf[4] === 0x66 && buf[5] === 0x72 && buf[6] === 0x65 && buf[7] === 0x65) ||
+		((buf[4] === 0x66 && buf[5] === 0x74 && buf[6] === 0x79 && buf[7] === 0x70) &&
+		(buf[8] === 0x71 && buf[9] === 0x74 && buf[10] === 0x20 && buf[7] === 0x20)) ||
+		(buf[4] === 0x6D && buf[5] === 0x64 && buf[6] === 0x61 && buf[7] === 0x74) ||
+		(buf[4] === 0x77 && buf[5] === 0x69 && buf[6] === 0x64 && buf[7] === 0x65)) {
```
Comment (sindresorhus) [inline index.js]: Rebase against master and you can now use the `{offset}` option in `check()` to handle these cases too.

**Comment by sindresorhus** [comment]
Would you be able to add some tests? With some fixtures.

File: index.js [hunk 4]
```diff
@@ -237,11 +237,10 @@ module.exports = input => {
...
 	if (check([0x0, 0x0, 0x0, 0x14, 0x66, 0x74, 0x79, 0x70, 0x71, 0x74, 0x20, 0x20]) ||
-		(buf[4] === 0x66 && buf[5] === 0x72 && buf[6] === 0x65 && buf[7] === 0x65) ||
-		((buf[4] === 0x66 && buf[5] === 0x74 && buf[6] === 0x79 && buf[7] === 0x70) &&
-		(buf[8] === 0x71 && buf[9] === 0x74 && buf[10] === 0x20 && buf[7] === 0x20)) ||
-		(buf[4] === 0x6D && buf[5] === 0x64 && buf[6] === 0x61 && buf[7] === 0x74) ||
-		(buf[4] === 0x77 && buf[5] === 0x69 && buf[6] === 0x64 && buf[7] === 0x65)) {
+		check([0x66, 0x72, 0x65, 0x65], {offset: 4}) ||
+		check([0x66, 0x74, 0x79, 0x70, 0x71, 0x74, 0x20, 0x20], {offset: 4}) ||
+		check([0x6D, 0x64, 0x61, 0x74], {offset: 4}) ||
+		check([0x77, 0x69, 0x64, 0x65], {offset: 4})) {
```
Comment (sindresorhus) [inline index.js]: Can you add an inline comment which of these are the `mjpeg` codec? For future reference.

Comment (karlhiramoto) [inline index.js]: Done.   I was trying to find others for tests, but hard to find ones that are both small, and no copyright issues.

**Comment by sindresorhus** [comment]
Great. Thank you! :)

**Pull request merged**